js/slideshow.js

```diff
@@ -187,6 +194,79 @@
 		},
 
 		/**
+		 * Calculates the luminance of an image
+		 *
+		 * @param {*} image
+		 * @param {string} mimeType
+		 *
+		 * @returns {string}
+		 * @private
+		 */
+		_calculateBackgroundColour: function (image, mimeType) {
+			var backgroundColour = this.darkBackgroundColour;
+			if (!this._isTransparent(image.mimeType)) {
+				return backgroundColour;
+			}
+
+			// The name has to be 'canvas'
+			var lumiCanvas = document.createElement('canvas');
+			lumiCanvas.width = 200;
+			lumiCanvas.height = 200;
+			var lumiCtx = lumiCanvas.getContext('2d');
+
+			lumiCanvas.height = lumiCanvas.width * (image.height / image.width);
+			lumiCtx.drawImage(image, 0, 0, lumiCanvas.width, lumiCanvas.height);
+			var imgData = lumiCtx.getImageData(0, 0, lumiCanvas.width, lumiCanvas.height);
+			var pix = imgData.data;
+			var pixelArraySize = pix.length;
+			var numberOfSamples = 4000; // Seems to be the sweet spot
+			var totalLuminance = 0;
+			var sampleNumber = 1;
+			var averageLuminance;
+			var totalAlpha = 0;
+			var alphaLevel;
+			var red = 0;
+			var green = 0;
+			var blue = 0;
+			var alpha = 0;
+			var lum = 0;
+
+			var sampleCounter = 0;
+
+			var sampleSize = Math.floor(pixelArraySize / numberOfSamples);
```

Comment:
I see two issues here:
* the actual number of samples is `numberOfSamples / 4` (because 4 colours for every pixel)
* `sampleSize` might be 0 if the image has less than `numberOfSamples` pixels

Suggestion:
```js
var numberOfSamples = 1000; // Seems to be the sweet spot
// ...
var itemsPerPixel = 4; // red, green, blue, alpha
var sampleSize =  Math.floor(pixelArraySize / (itemsPerPixel * numberOfSamples));
if (sampleSize <= 0){
  sampleSize = 1;
}
for (var i = 0, n = pixelArraySize; i < n; i += itemsPerPixel * sampleSize) {
```